src/core/utils/hooks/useCountdownTimer.ts

```diff
@@ -0,0 +1,41 @@
+import {useEffect, useRef, useState, useLayoutEffect} from "react";
+
+import {MINUTE_IN_S, SECOND_IN_MS} from "../time/timeConstants";
+import {CountDownResults} from "../time/timeTypes";
+import {calculateRemainingTime} from "../time/timeUtils";
+
+/**
+ * A React Hook that provides a countdown time
+ * @param {Date} target the target date we're counting down to
+ * @param {number} cadence the rate of the timer in seconds
+ *
+ * @returns {object} the CountDownResults object with ref
+ */
+function useCountDownTimer(target: Date, cadence = MINUTE_IN_S): CountDownResults {
```

Comment:
Let's require the cadence in MS which would be more consistent with the rest of the JS tools, such as setInterval. You will need to update the JSDoc above as well
```suggestion
function useCountDownTimer(target: Date, cadence = MINUTE_IN_MS): CountDownResults {
```